### ISSUE TITLE
Feature Request - Allow custom EBS volume size when Snapshot ID is provided

### ISSUE THREAD

**Issue by starchx** [aws/aws-parallelcluster#1629]
**Environment:**
 - aws-parallelcluster-2.5.1
 - Config:

```
[ebs custom1]
shared_dir = vol1
ebs_snapshot_id = snap-0071db2e5ebb2ff69 # snapshot is 20G.
volume_type = gp2
volume_size = 100
```

**Bug description and how to reproduce:**
volume_size is ignored. The volume created on Master instance is still 20G (based on snapshot size). Ideally a 100G volume is created and attached.

**Additional context:**
This is the part that defines the size for the EBS volume from the [source code](https://github.com/aws/aws-parallelcluster/blob/develop/cloudformation/ebs-substack.cfn.json):
```
        "Size": {
          "Fn::If": [
            "Vol1_UseEBSSnapshot",
            {
              "Ref": "AWS::NoValue"
            },
            {
              "Fn::If": [
                "Vol1_UseVolumeSize",
                {
                  "Fn::Select": [
                    "0",
                    {
                      "Ref": "VolumeSize"
                    }
                  ]
                },
                20
              ]
            }
          ]
        },
```
If changing to something like this, should work as expect:
```
            "Size": {
              "Fn::If": [
                "Vol1_UseVolumeSize",
                {
                  "Fn::Select": [
                    "0",
                    {
                      "Ref": "VolumeSize"
                    }
                  ]
                },
                {
                  "Ref": "AWS::NoValue"
                }
              ]
            },
```

I tested with CloudFormation, and it is possible to define a size even when snapshot is provided, for example, this will create an EBS volume with 100G.
```
Resources:
  NewVolume:
    Type: AWS::EC2::Volume
    Properties: 
        Size: 100
        AvailabilityZone: us-east-1a
        SnapshotId: snap-0071db2e5ebb2ff69
    DeletionPolicy: Delete
```
**Drawback:**
There is a potential drawback. If the size of snapshot provided is larger than the volume_size, the creation will fail.

**Comment by ganderaj** [comment]
Thanks for your contribution.

I find this feature request very useful. I do not see a reason why the parameter "volume_size" is set to be ignored when snapshot is provided. Users should be in a position to create volumes larger than their snapshots.

**Comment by ganderaj** [comment]
I have been working with AWS Parallel Cluster Team v2.5.1 to build a cluster for a GPU Intensive workloads. I have been using a snapshot of 4.5T in size to deploy a secondary EBS volume (to be shared across the compute nodes). As per the business needs we are looking for a 8T volume to be created from this snapshot. Below are the parameters that I have been using.
[ebs data]
ebs_snapshot_id  = snap-xxxxxxxxxx
shared_dir             = /data
volume_size          = 8192

However, I have observed that Parallel Cluster is ignoring the "volume_size" parameter when a snapshot id is included in the config file. With the above included config parameters, I am deploying a cluster with 4.5T secondary volume when the expected size is 8T. For interim purposes, to bypass this issue I am creating a snapshot with 8T size to be used in my config file. But this brings up our AWS Charges by $358 (8T-4.5T=3.5T snapshot costs) a month. Looking in a large scale use cases across an organization, this is a huge number. 

I request the Parallel Cluster DEV Team to look into this and try removing the explicit ignore in CFN Templates.

**Comment by ddeidda** [comment]
Yes, I think it's a feature that deserves to be considered. Marked as enhancement. Thanks for pointing it out!

**Comment by enrico-usai** [comment]
This feature has been implemented: https://github.com/aws/aws-parallelcluster/pull/4134 and will be part of the next release.

**Comment by gmarciani** [comment]
Feature included in [ParallelCluster 3.2.0](https://github.com/aws/aws-parallelcluster/releases/tag/v3.2.0)
